modules/custom/activity_logger/src/Service/ActivityLoggerFactory.php

```diff
@@ -97,20 +98,12 @@ public function createMessages(EntityBase $entity, $action) {
       // Set the values.
       $new_message['template'] = $message_type;
 
-      // The flagging entity does not implement getCreatedTime().
-      if ($entity->getEntityTypeId() === 'flagging') {
-        $new_message['created'] = $entity->get('created')->value;
-      }
-      else {
-        $new_message['created'] = $entity->getCreatedTime();
-      }
-
       // Get the owner or default to anonymous.
       if ($entity instanceof EntityOwnerInterface && $entity->getOwner() !== NULL) {
-        $new_message['uid'] = $entity->getOwner()->id();
+        $new_message['uid'] = (string) $entity->getOwner()->id();
       }
       else {
-        $new_message['uid'] = 0;
+        $new_message['uid'] = '0';
```

Comment:
Why the change to string instead of  the int type ?

Reply:
hello, it is used for "checkIfMessageExist", and $query->condition needs to have string|null as a parameter